mne/morph.py

```diff
@@ -0,0 +1,731 @@
+        It corresponds to FreeSurfer environment variable SUBJECTS_DIR.
+    subject_from : str | None
+        Name of the subject from which to morph as named in the SUBJECTS_DIR
+    subject_to : str | array | list of two arrays
+        Name of the subject on which to morph as named in the SUBJECTS_DIR
+        The default is 'fsaverage'. If morphing a surface source extimate,
+        subject_to can also be an array of vertices or a list of two arrays of
+        vertices to morph to. If morphing a volume source space, subject_to can
+        be the path to a MRI volume.
+    niter_affine : tuple of int
+        Number of levels (``niter_affine.__len__()``) and number per level of
+        iterations to perform the affine transform. Increasing index values
+        for the tuple mean later levels and each int represents the number
+        of iterations in that level.
+    niter_sdr : tuple of int
+        Number of levels (``niter_sdr.__len__()``) and number per level of
+        iterations to perform the sdr transform. Increasing index values
+        for the tuple mean later levels and each int represents the number
+        of iterations in that level.
+    grid_spacing : tuple
+        Voxel size of volume for each spatial dimension in mm.
+        If grid_spacing is None, MRIs won't be resliced. Note that in this case
+        both volumes must have the same number of slices in every
+        spatial dimension.
+    smooth : int | None
+        Number of iterations for the smoothing of the surface data.
+        If None, smooth is automatically defined to fill the surface
+        with non-zero values.
+    warn : bool
+        If True, warn if not all vertices were used.
+    xhemi : bool
+        Morph across hemisphere.
+
+    Notes
+    -----
+    .. versionadded:: X.X.X
+
+    See Also
+    --------
+    X
+
+    """
+
+    def __init__(self, src, subject_from=None, subject_to='fsaverage',
+                 subjects_dir=None, niter_affine=(100, 100, 10),
+                 niter_sdr=(5, 5, 3), grid_spacing=(5., 5., 5.), smooth=None,
+                 warn=True, xhemi=False, verbose=False):
+
+        # it's impossible to use the class without passing this check, so it
+        # only needs to be checked here
+        if not check_version('nibabel', '') or not check_version('dipy', ''):
+            raise ImportError(
+                'NiBabel (Python) and DiPy (Python) must be correctly '
+                'installed and accessible from Python')
+
+        # Set attributes
+        self.kind = None
+        self.subject_from = subject_from
+        self.subject_to = subject_to
+        self.subjects_dir = subjects_dir
+        self.niter_affine = niter_affine
+        self.niter_sdr = niter_sdr
+        self.grid_spacing = grid_spacing
+        self.smooth = smooth
+        self.warn = warn
+        self.xhemi = xhemi
+        self.morph_data = None
+
+        if src is None:
```

Comment:
so it will fail silently?

Reply:
necessary to create an "empty" object for loading